factory.go

```diff
@@ -40,7 +41,7 @@ type Parser interface {
 	NodeVersionsSupported() []string
 	ParseTransactions(ctx context.Context, txsData types.TxsData) (*types.TxsParsedResult, error)
 	ParseNativeEvents(ctx context.Context, eventsData types.EventsData) (*types.EventsParsedResult, error)
-	ParseEthLogs(ctx context.Context, eventsData types.EventsData) (*types.EventsParsedResult, error)
+	ParseEthLogs(ctx context.Context, cache zcache.ZCache, eventsData types.EventsData) (*types.EventsParsedResult, error)
```

Comment:
There is no need to pass the cache here, as it is actually passed when the NewParser is created. We should use that same cache!